### ISSUE TITLE
A lot of tests fail when trying to compile

### ISSUE THREAD

**Issue by Niko-sk2x** [Deamon5550/Despector#8]
```
:test

org.spongepowered.test.kotlin.StringTests > testSimpleStringTemplate FAILED
    org.junit.ComparisonFailure at StringTests.java:66

org.spongepowered.test.kotlin.BasicTests > testNoVoidReturnDeclared FAILED
    org.junit.ComparisonFailure at BasicTests.java:78

org.spongepowered.test.kotlin.BasicTests > testLocals FAILED
    org.junit.ComparisonFailure at BasicTests.java:108

org.spongepowered.test.kotlin.BasicTests > testSimple FAILED
    org.junit.ComparisonFailure at BasicTests.java:55

org.spongepowered.test.kotlin.TernaryTests > testSimpleTernary FAILED
    org.junit.ComparisonFailure at TernaryTests.java:63

org.spongepowered.test.kotlin.IfTests > testSimpleIf FAILED
    org.junit.ComparisonFailure at IfTests.java:67

org.spongepowered.test.kotlin.RangeTests > testIfRange FAILED
    org.junit.ComparisonFailure at RangeTests.java:84

org.spongepowered.test.kotlin.OperatorTests > testAdd FAILED
    org.junit.ComparisonFailure at OperatorTests.java:101

org.spongepowered.test.kotlin.OperatorTests > testDiv FAILED
    org.junit.ComparisonFailure at OperatorTests.java:173

org.spongepowered.test.kotlin.OperatorTests > testRem FAILED
    org.junit.ComparisonFailure at OperatorTests.java:197

org.spongepowered.test.kotlin.OperatorTests > testShl FAILED
    org.junit.ComparisonFailure at OperatorTests.java:269

org.spongepowered.test.kotlin.OperatorTests > testShr FAILED
    org.junit.ComparisonFailure at OperatorTests.java:221

org.spongepowered.test.kotlin.OperatorTests > testSub FAILED
    org.junit.ComparisonFailure at OperatorTests.java:125

org.spongepowered.test.kotlin.OperatorTests > testUshr FAILED
    org.junit.ComparisonFailure at OperatorTests.java:245

org.spongepowered.test.kotlin.OperatorTests > testFloatCompare FAILED
    org.junit.ComparisonFailure at OperatorTests.java:301

org.spongepowered.test.kotlin.OperatorTests > testIntConstant FAILED
    org.junit.ComparisonFailure at OperatorTests.java:55

org.spongepowered.test.kotlin.OperatorTests > testMultiply FAILED
    org.junit.ComparisonFailure at OperatorTests.java:149

org.spongepowered.test.kotlin.OperatorTests > testNegative FAILED
    org.junit.ComparisonFailure at OperatorTests.java:77

org.spongepowered.test.decompile.SwitchTests > testReturnSwitchEclipse FAILED
    org.junit.ComparisonFailure at SwitchTests.java:258

org.spongepowered.test.decompile.MethodTests > testNewArray FAILED
    org.junit.ComparisonFailure at MethodTests.java:123

org.spongepowered.test.ast.LambdaTest > testProducer FAILED
    org.junit.ComparisonFailure at LambdaTest.java:77

org.spongepowered.test.ast.LambdaTest > testMethodRef FAILED
    org.junit.ComparisonFailure at LambdaTest.java:89

org.spongepowered.test.ast.LambdaTest > testConsumer FAILED
    org.junit.ComparisonFailure at LambdaTest.java:65

org.spongepowered.test.ast.LambdaTest > testLambda FAILED
    org.junit.ComparisonFailure at LambdaTest.java:53

122 tests completed, 24 failed
:test FAILED
```

**Comment by Niko-sk2x** [comment]
Seems like this commit https://github.com/Despector/Despector/commit/df5857c2f86a04b9a5154703eba373737011ccec broke handling of primitive type array signatures - `LI;` is not a valid signature.

**Comment by Deamon5550** [comment]
new array was fixed in 9c091d4cb75a146de45c816c0cb383381db81a93
the lambda issues in f4e1c39eda738384f2552fcd20d8b98fc78a1920
Two are syntax issues that aren't actually errors and the rest are an issue with the kotlin locals handler that needs a complete rewrite to better filter all of the extra locals that kotlin creates.

**Comment by Deamon5550** [comment]
Rest were fixed in 370be4ff5f6e0676841f44f64eb75bf81bd1994a 58727cccd23dc415855c1a912d66b6e2b30adcbd
